pkg/utils/media_serialization.go

```diff
@@ -34,17 +35,26 @@ import (
 */
 
 func SerializeMediaForRelay(w io.Writer, data []byte, ts time.Duration) error {
-	err := binary.Write(w, binary.BigEndian, ts)
+	// Only 1 write for a single parsable unit
+
+	b := &bytes.Buffer{}
+
+	err := binary.Write(b, binary.BigEndian, ts)
```

Comment:
Not critical, but `binary.Write` is kind of slow. Ideally calling `binary.BigEndian.PutUint64` would be better. But it's fine if it's not on a hot path.